packages/block-editor/src/hooks/typography.js

```diff
@@ -76,159 +42,93 @@ export const TYPOGRAPHY_SUPPORT_KEYS = [
 	LETTER_SPACING_SUPPORT_KEY,
 ];
 
-export function TypographyPanel( props ) {
-	const { clientId } = props;
-	const isFontFamilyDisabled = useIsFontFamilyDisabled( props );
-	const isFontSizeDisabled = useIsFontSizeDisabled( props );
-	const isFontAppearanceDisabled = useIsFontAppearanceDisabled( props );
-	const isLineHeightDisabled = useIsLineHeightDisabled( props );
-	const isTextDecorationDisabled = useIsTextDecorationDisabled( props );
-	const isTextTransformDisabled = useIsTextTransformDisabled( props );
-	const isLetterSpacingDisabled = useIsLetterSpacingDisabled( props );
+function TypographyInspectorControl( { children } ) {
+	return (
+		<InspectorControls __experimentalGroup="typography">
+			{ children }
+		</InspectorControls>
+	);
+}
 
-	const hasFontStyles = ! useIsFontStyleDisabled( props );
-	const hasFontWeights = ! useIsFontWeightDisabled( props );
+export function TypographyPanel( {
+	clientId,
+	name,
+	attributes,
+	setAttributes,
+} ) {
+	const settings = {
+		typography: {
+			fontFamilies: {
+				custom: useSetting( 'typography.fontFamilies' ),
```

Comment:
How come we can't do one `useSetting( 'typography' )` call?

Reply:
I tried to map exactly what we used to do before the PR. I think if we do "typography" we'd be losing some backward compatibility code that might trigger for old block editor settings.